grails-app/controllers/com/unifina/controller/api/AuthApiController.groovy

```diff
@@ -171,9 +166,8 @@ class AuthApiController {
 			return render([success: false, error: userService.beautifyErrors(command.errors.getAllErrors())] as JSON)
 		}
 
-		String salt = saltSource instanceof NullSaltSource ? null : registrationCode.username
 		RegistrationCode.withTransaction { status ->
-			user.password = springSecurityService.encodePassword(command.password, salt)
+			user.password = passwordEncoder.encodePassword(command.password, salt)
```

Comment:
@kare **major finding, looks broken:**
- `salt` is not defined
- `encodePassword(String, String)` is not defined. Correct method signature is `encodePassword(String password)`.